viabel/objectives.py

```diff
@@ -133,6 +141,10 @@ def variational_objective(var_param):
             return -lower_bound
         self._objective_and_grad = value_and_grad(variational_objective)
 
+    def _hessian_vector_product(self, var_param, x):
+        hvp = make_hvp(self.variational_objective)(var_param)[0]
```

Comment:
You shouldn't need to call `make_hvp` every time. Just do it once in `__init__` and then reuse